hardware/kubernetes_client_test.go

```diff
@@ -0,0 +1,127 @@
+package hardware_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/tinkerbell/hegel/hardware"
+	tinkv1alpha1 "github.com/tinkerbell/tink/pkg/apis/core/v1alpha1"
+	tink "github.com/tinkerbell/tink/pkg/controllers"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestKubernetesClientLists(t *testing.T) {
+	const ip = "10.0.10.0"
+	const name = "hello-world"
+
+	listerClient := &ListerClientMock{}
+	listerClient.
+		On("List", mock.Anything, mock.Anything, mock.Anything).
+		Run(func(args mock.Arguments) {
+			require.IsType(t, &tinkv1alpha1.HardwareList{}, args.Get(1))
+			hw := args.Get(1).(*tinkv1alpha1.HardwareList)
+
+			hw.Items = append(hw.Items, tinkv1alpha1.Hardware{
+				ObjectMeta: v1.ObjectMeta{Name: name},
+				Spec: tinkv1alpha1.HardwareSpec{
+					Metadata: &tinkv1alpha1.HardwareMetadata{
+						Facility: &tinkv1alpha1.MetadataFacility{},
+						Instance: &tinkv1alpha1.MetadataInstance{
+							OperatingSystem: &tinkv1alpha1.MetadataInstanceOperatingSystem{},
+							Ips:             []*tinkv1alpha1.MetadataInstanceIP{},
+						},
+					},
+				},
+			})
+		}).
+		Return((error)(nil))
+
+	client := hardware.NewKubernetesClientWithClient(listerClient)
+
+	_, err := client.ByIP(context.Background(), ip)
+	require.NoError(t, err)
+
+	require.Equal(t, len(listerClient.Calls), 1)
+	args := listerClient.Calls[0].Arguments
+
+	_, ok := args.Get(1).(*tinkv1alpha1.HardwareList)
+	assert.True(t, ok)
+
+	opts := args.Get(2).([]crclient.ListOption)
+	require.Len(t, opts, 1)
+
+	matchingFields, ok := opts[0].(crclient.MatchingFields)
+	require.True(t, ok)
+
+	require.Contains(t, matchingFields, tink.HardwareIPAddrIndex)
+	assert.Equal(t, ip, matchingFields[tink.HardwareIPAddrIndex])
+
+	// todo(chrisdoherty4) Validate the returned hardware Export() has correctly serialized data.
```

Comment:
Would a short string comparison should work for this test?

Reply:
I suspect so. I'll put something together.

Reply:
Added